### PR TITLE
Add `combobox` inline autocomplete example

### PR DESCRIPTION
**Pull request by cloud-walker**
Trying to complete: #939 "Combobox with autoComplete set to inline"

I'm not sure if the test cases are enough and if the example is complete enough, let me know!

### CODE REVIEW

File: packages/ariakit/src/combobox/__examples__/combobox-inline-autocomplete/test.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  render(<Example />);
+  const combobox = getCombobox();
+  const popover = getPopover();
+  await press.Tab();
+  expect(popover).not.toBeVisible();
+  expect(combobox).toHaveFocus();
+  await type("a");
+  expect(popover).toBeVisible();
+  await press.ArrowDown();
+  expect(combobox).toHaveValue("Apple");
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-inline-autocomplete/test.tsx]: Some examples may re-mount their elements in future changes (for example, if we start showing the popover conditionally, the element references will be different each time the popover opens), so it's better to query the elements again on each assertion (an exception would be an extensive test suite where multiple queries could impact performance significantly and slow down the test, which is often not the case).

```suggestion
  await press.Tab();
  expect(getPopover()).not.toBeVisible();
  expect(getCombobox()).toHaveFocus();
  await type("a");
  expect(getPopover()).toBeVisible();
  await press.ArrowDown();
  expect(getCombobox()).toHaveValue("Apple");
```

File: packages/ariakit/src/combobox/__examples__/combobox-inline-autocomplete/test.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  render(<Example />);
+  const combobox = getCombobox();
+  const popover = getPopover();
+  await press.Tab();
+  expect(popover).not.toBeVisible();
+  expect(combobox).toHaveFocus();
+  await type("w");
+  expect(popover).toBeVisible();
+  await press.ArrowUp();
+  expect(combobox).toHaveValue("Watermelon");
```
Comment (diegohaz) [inline packages/ariakit/src/combobox/__examples__/combobox-inline-autocomplete/test.tsx]:
```suggestion
  render(<Example />);
  await press.Tab();
  expect(getPopover()).not.toBeVisible();
  expect(getCombobox()).toHaveFocus();
  await type("w");
  expect(getPopover()).toBeVisible();
  await press.ArrowUp();
  expect(getCombobox()).toHaveValue("Watermelon");
```

File: packages/ariakit/src/combobox/__examples__/combobox-inline-autocomplete/test.tsx (hunk 1 above)
Comment (cloud-walker) [reply packages/ariakit/src/combobox/__examples__/combobox-inline-autocomplete/test.tsx]: done

**Review by diegohaz — APPROVED** [review]
Nice! Thank you!

**Pull request merged**